### PR TITLE
[WIP/PoC] Attempt to use the fasteners library

### PR DESCRIPTION
**Pull request by dstufft**

### CODE REVIEW

File: cachecontrol/caches/file_cache.py [hunk 1]
```diff
@@ -79,13 +100,21 @@ def _fn(self, name):
         parts = list(hashed[:5]) + [hashed]
         return os.path.join(self.directory, *parts)
 
+    def _lockname(self, name):
+        if isinstance(self.lock_class, PYLockFileBase):
+            return self.name
```
Comment (dsully) [inline cachecontrol/caches/file_cache.py]: This should be 'name', not 'self.name'

File: cachecontrol/caches/file_cache.py [hunk 2]
```diff
@@ -79,13 +100,21 @@ def _fn(self, name):
...
+        else:
+            return self.name + ".lock"
```
Comment (dsully) [inline cachecontrol/caches/file_cache.py]: Same here.

File: cachecontrol/caches/file_cache.py [hunk 3]
```diff
@@ -79,13 +100,21 @@ def _fn(self, name):
...
+
     def get(self, key):
         name = self._fn(key)
-        if not os.path.exists(name):
-            return None
 
-        with open(name, 'rb') as fh:
-            return fh.read()
+        with self.lock_class(self._lockname(name)):
+            if not os.path.exists(name):
```
Comment (dsully) [inline cachecontrol/caches/file_cache.py]: Should this be refactored as a try/except block, instead of testing for path existence?

**Comment by kloczek** [comment]
To be able test that PR it would be necessary to modify module metadata to statr require on install time `fasteners` instead `lockfile`.
Is it possible to add that part ot the PR as well?

**Pull request closed without merging**